infra/modules/providers/azure/app-service/README.md

```diff
@@ -0,0 +1,60 @@
+## Azure App Service
+
+Build, deploy, and scale enterprise-grade web, mobile, and API apps running on any platform. Meet rigorous performance, scalability, security and compliance requirements while using a fully managed platform to perform infrastructure maintenance.
+
+More information for Azure App Services can be found [here](https://azure.microsoft.com/en-us/services/app-service/)
+
+Cobalt gives ability to specify following settings for App Service based on the requirements:
+- name : The name of the App Services to be created.
+- service_plan_resource_group_name : The Name of the Resource Group where the Service Plan exists.
+- app_service_plan_id : The ID of the App Service Plan within which the App Service exists is populated automatically based on service plan.
+- tags : A mapping of tags to assign to the resource.
+- app_settings : A key-value pair of App Settings. Settings for private Container Registries
+  - DOCKER_REGISTRY_SERVER_URL : The docker registry server URL for app service to be created
+  - DOCKER_REGISTRY_SERVER_USERNAME : The docker registry server usernames for app services to be created
+  - DOCKER_REGISTRY_SERVER_PASSWORD : The docker registry server passwords for app services to be created
+- site_config : 
+  - linux_fx_version : Linux App Frameworks and versions for the App Services created. Possible options are a Docker container (DOCKER|<user/image:tag>), a base-64 encoded Docker Compose file (COMPOSE|${base64encode(file("compose.yml"))}) or a base-64 encoded Kubernetes Manifest (KUBE|${base64encode(file("kubernetes.yml"))}).
+  - always_on : Should the app be loaded at all times? Defaults to false.
+
+Please click the [link](https://www.terraform.io/docs/providers/azurerm/d/app_service.html) to get additional details on settings in Terraform for Azure App Service.
+
+## Usage
+
+```
+data "azurerm_resource_group" "appsvc" {
+  name      = "${var.service_plan_resource_group_name}"
+}
+
+data "azurerm_app_service_plan" "appsvc" {
+  name                    = "${var.service_plan_name}"
+  resource_group_name     = "${data.azurerm_resource_group.appsvc.name}"
+}
+
+resource "azurerm_app_service" "appsvc" {
+  name                = "${var.app_service_name[count.index]}"
+  resource_group_name = "${data.azurerm_resource_group.appsvc.name}"
+  location            = "${data.azurerm_resource_group.appsvc.location}"
+  app_service_plan_id = "${data.azurerm_app_service_plan.appsvc.id}"
+  tags                = "${var.resource_tags}"
+  count               = "${length(var.app_service_name)}"
+
+  app_settings {
+    DOCKER_REGISTRY_SERVER_URL      = "${var.docker_registry_server_url[count.index]}"
+    DOCKER_REGISTRY_SERVER_USERNAME = "${var.docker_registry_server_username[count.index]}"
+    DOCKER_REGISTRY_SERVER_PASSWORD = "${var.docker_registry_server_password[count.index]}"
+  }
+
+  site_config {
+    linux_fx_version = "${var.site_config_linux_fx_version[count.index]}"
```

Comment:
Where are these variables defined?

Reply:
Variables are defined in variables.tf file

Reply:
both the app service name and container exec command should be driven off the same variable.

Reply:
In an ideal situation yes but that seems too rigid to push developers to give same names. Many containers will have additional versions that might not be part of the app service name.

Reply:
ok.... if we go with multiple variables, how are you accounting for situations where there's a array length mismatch between `site_config_linux_fx_version` and `app_service_name` variables?